docs/design-documents/hal/0000-spi-overhaul.md

```diff
@@ -98,14 +98,15 @@ typedef struct {
      *  testing.
      */
     uint32_t    maximum_frequency;
-    /** Each bit represents the corresponding symbol length. lsb => 1bit, msb => 32bits.
-     *
-     * For example, if the peripheral supports 8 bits, 12bits to 16bits and 32bits,
-     * the value should be 0x8000F880.
-     */
-    uint32_t    symbol_length;
+    /** Each bit represents the corresponding word length. lsb => 1bit, msb => 32bit. */
+    uint32_t    word_length;
+    uint16_t    slave_delay_between_symbols_ns; /**< specifies required number of ns between transmission of successive symbols in slave mode. */
```

Comment:
Would this be more appropriate as `maximum_effective_frequency`? This is the fastest a SPI master using DMA will be able to send data to a slave without losing bytes.

Reply:
`maximum_effective_frequency` would be the same as `maximum_frequency` for spi slave. 

The problem is that our implementation for synchronous transfer doesn't use DMA. On `K66F` we got TX, RX fifos and there is no problem on the slave side to handle max frequency reported by device capabilities (no delay between symbols is required). 
On `NUCLEO_F429ZI` symbols are transmitted one by one by writing/reading successive symbols to/from the TX/RX registers. Because of that max effective frequency for the slave is ~1 MHz. I wanted to set `maximum_frequency` for the slave to 1 MHz, but this is not consistent with the following requirement:
`The supported frequency range must include the range [0.2..2] MHz.`